zerver/lib/user_groups.py

```diff
@@ -103,6 +103,13 @@ def has_user_group_access(
     if user_group.is_system_group:
         return False
 
+    # This is a temporary exception and this should be removed as soon
+    # as `group_creator` is set as a default for `can_manage_group`
+    # property of user groups. See this topic for more details:
+    # https://chat.zulip.org/#narrow/stream/3-backend/topic/Group.20creation.20-.20who.20can.20change.20the.20setting.2E/near/1943861
+    if user_group.creator and user_group.creator.id == user_profile.id:
+        return True
+
     can_edit_all_user_groups = user_profile.can_edit_all_user_groups()
 
     group_member_ids = get_user_group_direct_member_ids(user_group)
```

Comment:
I think we should remove this check for checking if the user is member of the group, since the setting is about managing all the user groups, so if the setting is set to allow members they should be allowed to manage all the user groups. I think this member check was to prevent members having ability to edit all the groups, but that should be doable now by setting this to "Moderators" and then using the group-level setting.